src/sequence_manager.cpp

```diff
@@ -44,7 +46,9 @@ bool SequenceManager::sequenceExists(const uint64_t sequenceId) const {
     return sequences.count(sequenceId);
 }
 
-Status SequenceManager::removeTimedOutSequences(std::chrono::steady_clock::time_point currentTime) {
+Status SequenceManager::removeTimedOutSequences() {
+    std::unique_lock<std::mutex> sequenceManagerLock(mutex);
+    std::chrono::steady_clock::time_point currentTime = std::chrono::steady_clock::now();
     for (auto it = sequences.cbegin(); it != sequences.cend();) {
         auto& sequence = it->second;
         auto timeDiff = currentTime - sequence.getLastActivityTime();
```

Comment:
Here we access sequence properties. Moreover, in the next line we remove the sequence from sequences container. 
What if there is ongoing inference on that sequence? Locking sequence manager will not prevent removal of sequence that is being used.